src/Voron/Data/Fixed/FixedSizeTree.cs

```diff
@@ -35,13 +35,22 @@ public unsafe partial class FixedSizeTree : IDisposable
         private readonly int _maxEmbeddedEntries;
 
         private NewPageAllocator _newPageAllocator;
-        private RootObjectType? _type;
         private FastStack<FixedSizeTreePage> _cursor;
         private int _changes;
 
         public LowLevelTransaction Llt => _tx;
 
-        internal RootObjectType? Type => _type;
+        internal RootObjectType? Type
+        {
+            get
+            {
+                var header = _parent.DirectRead(_treeName);
```

Comment:
Is there an option to calculate that only when we are switching from one RootObjectType to another?

Reply:
We did that, but the issue is that we may have multiple _instances_ of this pointing to the same location, that is why we have to read it anew every time